CesiumIonClient/src/LoginToken.cpp

```diff
@@ -0,0 +1,51 @@
+#include <CesiumClientCommon/JwtTokenUtility.h>
+#include <CesiumIonClient/LoginToken.h>
+#include <CesiumUtility/ErrorList.h>
+#include <CesiumUtility/JsonHelpers.h>
+#include <CesiumUtility/Result.h>
+
+#include <rapidjson/document.h>
+
+#include <chrono>
+#include <cstdint>
+#include <string>
+#include <utility>
+
+using namespace CesiumUtility;
+
+namespace CesiumIonClient {
+CesiumUtility::Result<LoginToken>
+LoginToken::parse(const std::string& tokenString) {
+  CesiumUtility::Result<rapidjson::Document> payloadResult =
+      CesiumClientCommon::JwtTokenUtility::parseTokenPayload(tokenString);
+  if (!payloadResult.value) {
+    return Result<LoginToken>(std::move(payloadResult.errors));
+  }
+
+  rapidjson::Document& json = *payloadResult.value;
+  int64_t expires = JsonHelpers::getInt64OrDefault(json, "exp", -1);
+
+  return Result<LoginToken>(LoginToken(tokenString, expires));
+}
+
+bool LoginToken::isValid() const {
+  if (this->_expires == -1) {
+    // If the value is -1, that suggests we have no expiration date, so the
+    // token is always valid.
+    return true;
+  }
```

Comment:
This feels hand-wavey since there's nothing alluding to an optional expiration time in the `LoginToken` constructor. Maybe consider _all_ negative numbers as no expiration date, or otherwise make this number a `std::optional`?